checks/evaluation/pinned_dependencies.go

```diff
@@ -139,17 +133,22 @@ func PinningDependencies(name string, c *checker.CheckRequest,
 		return checker.CreateRuntimeErrorResult(name, err)
 	}
 
-	// Scores may be inconclusive.
-	actionScore = maxScore(0, actionScore)
-	dockerFromScore = maxScore(0, dockerFromScore)
-	dockerDownloadScore = maxScore(0, dockerDownloadScore)
-	scriptScore = maxScore(0, scriptScore)
-	pipScore = maxScore(0, pipScore)
-	npmScore = maxScore(0, npmScore)
-	goScore = maxScore(0, goScore)
+	// If no dependencies of an ecossystem are found, it results in an inconclusive score.
+	// We filter out inconclusive scores so only applicable ecossystems are considered in
+	// the final score.
+	scores := []int{actionScore, dockerFromScore, dockerDownloadScore, scriptScore, pipScore, npmScore, goScore}
+	conclusiveScores := []int{}
+	for i := range scores {
+		if scores[i] != checker.InconclusiveResultScore {
+			conclusiveScores = append(conclusiveScores, scores[i])
+		}
+	}
+
+	if len(conclusiveScores) == 0 {
+		return checker.CreateInconclusiveResult(name, "no dependencies found")
+	}
 
-	score := checker.AggregateScores(actionScore, dockerFromScore,
-		dockerDownloadScore, scriptScore, pipScore, npmScore, goScore)
+	score := checker.AggregateScores(conclusiveScores...)
```

Comment:
does this contradict the approach from the issue?
https://github.com/ossf/scorecard/issues/3254#issuecomment-1692422889

fixing this could fix the issue above about forgetting ecosystems

Reply:
Yes as replied above, but we can make the evaluation changes in this PR instead of in a second PR if you'd like.